### PR TITLE
[bitnami/keycloak] add annotation for auto-generated secrets

### PR DESCRIPTION
**Pull request by aabouzaid**
### Description of the change

Allow templating `annotation` in `auth` and `externalDatabase`.

### Benefits

Avoid auto-generation of the secrets by passing [Helm hooks](https://helm.sh/docs/topics/charts_hooks/), namely, `"helm.sh/hook": pre-install` so no need to extract passwords during the upgrade since the password will not change anyway.

### Possible drawbacks

None.

### Applicable issues

None.

### Additional information

**Background:**
Since the PostgreSQL chart is used as dependency, during the upgrade, you need to extract the passwords from the secrets because those secrets are seeded in the PVC, as shown in the [Credential errors while upgrading chart releases](https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases).

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by javsalgar — APPROVED** [review]

File: bitnami/keycloak/templates/secrets.yaml [hunk 1]
```diff
@@ -11,8 +11,8 @@ metadata:
   namespace: {{ include "common.names.namespace" . | quote }}
   labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
     app.kubernetes.io/component: keycloak
-  {{- if .Values.commonAnnotations }}
-  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- if or .Values.commonAnnotations .Values.auth.annotations }}
+  annotations: {{- include "common.tplvalues.merge" ( dict "values" (list .Values.commonAnnotations .Values.auth.annotations) "context" $ ) | nindent 4 }}
```
Comment (andresbono) [inline bitnami/keycloak/templates/secrets.yaml:15]: Same suggestion than above

```suggestion
  annotations: {{- include "common.tplvalues.merge" ( dict "values" (list .Values.auth.annotations .Values.commonAnnotations) "context" $ ) | nindent 4 }}
```

File: bitnami/keycloak/templates/secret-external-db.yaml [hunk 2]
```diff
@@ -10,8 +10,8 @@ metadata:
   name: {{ printf "%s-externaldb" .Release.Name }}
   namespace: {{ .Release.Namespace | quote }}
   labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
-  {{- if .Values.commonAnnotations }}
-  annotations: {{- include "common.tplvalues.render" (dict "value" .Values.commonAnnotations "context" $) | nindent 4 }}
+  {{- if or .Values.commonAnnotations .Values.externalDatabase.annotations }}
+  annotations: {{- include "common.tplvalues.merge" ( dict "values" (list .Values.commonAnnotations .Values.externalDatabase.annotations) "context" $ ) | nindent 4 }}
```
Comment (andresbono) [inline bitnami/keycloak/templates/secret-external-db.yaml:14]: This is just a detail, but wouldn't it make more sense to swap the order in the objects to merge so the `externalDatabase.annotations` (more specific) take precedence over the common ones?


```suggestion
  annotations: {{- include "common.tplvalues.merge" ( dict "values" (list .Values.externalDatabase.annotations .Values.commonAnnotations) "context" $ ) | nindent 4 }}
```

**Review by andresbono — COMMENTED** [review]
Thank you for your contribution! Could you take a look at my comments?

File: bitnami/keycloak/templates/secret-external-db.yaml (hunk 2 above)
Comment (aabouzaid) [reply bitnami/keycloak/templates/secret-external-db.yaml]: Fixed :+1:

File: bitnami/keycloak/templates/secrets.yaml (hunk 1 above)
Comment (aabouzaid) [reply bitnami/keycloak/templates/secrets.yaml]: Fixed :+1:

**Review by andresbono — APPROVED** [review]
LGTM, Thank you!!

**Review by andresbono — APPROVED** [review]

**Pull request merged**